common/config/src/raw.rs

```diff
@@ -1,12 +1,75 @@
 use crate::error::ConfigError;
 use crate::{argv, get_hostname, properties};
 use http::types::params::Params;
+use humanize_rs::bytes::Bytes;
+use serde::de::{Deserializer, Error, Unexpected, Visitor};
 use serde::{Deserialize, Serialize};
+use std::fmt;
 use std::fs::File;
 use std::io::{Seek, SeekFrom};
 use std::path::{Path, PathBuf};
 use std::vec::Vec;
 
+pub fn filesize_deser<'de, D>(d: D) -> Result<Option<u64>, D::Error>
```

Comment:
Ouch, manually implementing serde serializers/deserializers is never fun.

Reply:
I tried using https://crates.io/crates/serde-humanize-rs but it makes the assumption that you're going to represent sizes as `usize` but the fs APIs use u64 for all file sizes. It was either roll our own serde deserializer or put in a bunch of code to handle loss of precision in converting from `usize` to `u64`. This seemed less problematic.

Reply:
Huh, interesting, usize to u64 shouldn't be lossy on any modern system, can't think of anything with a 128bit pointer. We could probably handle it with TryFrom and a LossOfPrecision error variant, but this is as tidy or tidier.